Assets/Tests/InputSystem/CorePerformanceTests.cs

```diff
@@ -1100,4 +1102,106 @@ public void Performance_OptimizedControls_ReadingPose4kTimes(OptimizationTestTyp
     }
 
 #endif
+
+    #if UNITY_2022_3_OR_NEWER
+    [PrebuildSetup(typeof(ProjectWideActionsBuildSetup))]
+    [PostBuildCleanup(typeof(ProjectWideActionsBuildSetup))]
+    [UnityTest, Performance]
+    [Category("Performance")]
+    public IEnumerator Performance_MeasureInputSystemFrameTimeWithProfilerMarkers_FPS()
+    {
+        string[] markers =
+        {
+            "InputUpdate",
+            "InputSystem.onBeforeUpdate",
+            "InputSystem.onAfterUpdate",
+            "PreUpdate.NewInputUpdate",
+            "PreUpdate.InputForUIUpdate",
+            "FixedUpdate.NewInputFixedUpdate"
+        };
+
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        var mouse = InputSystem.AddDevice<Mouse>();
+
+        var moveAction = InputSystem.actions.FindAction("Move");
+        var lookAction = InputSystem.actions.FindAction("Look");
+        var attackAction = InputSystem.actions.FindAction("Attack");
+        var jumpAction = InputSystem.actions.FindAction("Jump");
+        var sprintAction = InputSystem.actions.FindAction("Sprint");
+
+        int performedCallCount = 0;
+
+        moveAction.performed += context => {
+            performedCallCount++;
+        };
+
+        lookAction.performed += context => {
+            performedCallCount++;
+        };
+
+        attackAction.performed += context => {
+            performedCallCount++;
+        };
+
+        jumpAction.performed += context => {
+            performedCallCount++;
+        };
+
+        sprintAction.performed += context => {
+            performedCallCount++;
+        };
+
+        using (Measure.ProfilerMarkers(markers))
+        {
+            for (int i = 0; i < 500; ++i)
+            {
+                PressAndRelease(keyboard.wKey, queueEventOnly: true);
+                PressAndRelease(keyboard.aKey, queueEventOnly: true);
+                PressAndRelease(keyboard.sKey, queueEventOnly: true);
+                PressAndRelease(keyboard.dKey, queueEventOnly: true);
+
+                PressAndRelease(keyboard.leftShiftKey, queueEventOnly: true);
+
+                PressAndRelease(keyboard.spaceKey, queueEventOnly: true);
+
+                Click(mouse.leftButton, queueEventOnly: true);
+
+                //mouse movements for higher polling mice (66*60 ~ 4kHz)
```

Comment:
I think the for loop here should match the achievable event rate at the OS. So 66 here like you are using means 66 Hz. I think 500-1000 Hz is common as hardware sampling frequency. The maximum I have measured on my machine with a 8 kHZ mouse have been 5700 mouse events per second. Which means 95 mouse events per frame @60 Hz. So that would map to 95 here in this for. The outer for loop just means, lest simulate 500 frames.

Reply:
https://github.com/Unity-Technologies/InputSystem/pull/2027/commits/79525e045620aef6636af2f751375e40b02f6862